### PR TITLE
Add profile names to thread picker, search profile names

### PR DESCRIPTION
**Pull request by michaelkirk**
PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/ViewControllers/SelectThreadViewController.m [hunk 1]
```diff
@@ -131,11 +132,13 @@ - (void)updateTableContents
     __weak SelectThreadViewController *weakSelf = self;
     ContactsViewHelper *helper = self.contactsViewHelper;
     OWSTableContents *contents = [OWSTableContents new];
-    OWSTableSection *section = [OWSTableSection new];
 
-    // Threads
+    // Existing threads are listed first, ordered by most recently active
+    OWSTableSection *recentChatsSection = [OWSTableSection new];
+    recentChatsSection.headerTitle = NSLocalizedString(
+        @"SELECT_THREAD_TABLE_RECENT_CHATS_TITLE", @"Table section header for recently active conversations");
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/SelectThreadViewController.m]: Just spitballing here: I wonder if we should interleave these search results (visually distinguishing contacts from conversations somehow) and sort them by "match weight" where, for example, matches at the start of a name would be weighted more heavily.

e.g. searching for "Bo", a user would see "Bob James <contact>" before "Trip to Boulder <group>" and a conversation with "Bob James <1:1>" before "Alice Arecibo <contact>".

File: Signal/src/util/Searcher.swift [hunk 2]
```diff
@@ -3,6 +3,91 @@
 //
 
 import Foundation
+import SignalServiceKit
+
+@objc
+class ConversationSearcher: NSObject {
```
Comment (charlesmchen) [inline Signal/src/util/Searcher.swift]: Should each Searcher have its own source file?

File: SignalMessaging/utils/DebugLogger.m [hunk 3]
```diff
@@ -106,7 +106,7 @@ - (void)enableTTYLogging
     // This should be redundant with the logic above.
     [logPathSet addObjectsFromArray:self.fileLogger.logFileManager.unsortedLogFilePaths];
     NSArray<NSString *> *logPaths = logPathSet.allObjects;
-    return [logPaths sortedArrayUsingSelector:@selector(compare:)];
+    return [logPaths sortedArrayUsingSelector:@selector((compare:))];
```
Comment (charlesmchen) [inline SignalMessaging/utils/DebugLogger.m]: Huh!  Why was this necessary?

Comment (charlesmchen) [reply SignalMessaging/utils/DebugLogger.m]: Is this to suppress / work around the "multiple matching methods with varying types" warning?

File: Signal/src/util/Searcher.swift (hunk 2 above)
Comment (michaelkirk) [reply Signal/src/util/Searcher.swift]: Sure, I'll move ConversationSearcher into it's own file. I'm going to leave AnySearcher with Searcher since it's just a thing wrapper for objc compatibility.

File: SignalMessaging/utils/DebugLogger.m (hunk 3 above)
Comment (michaelkirk) [reply SignalMessaging/utils/DebugLogger.m]: It was - I can't imagine adding those parentheses actually makes it safer though, probably just snuck past the linter.

File: Signal/src/ViewControllers/SelectThreadViewController.m (hunk 1 above)
Comment (michaelkirk) [reply Signal/src/ViewControllers/SelectThreadViewController.m]: Happy to discuss this more for a future iteration.
